app/main/views.py

```diff
@@ -49,35 +49,101 @@ def list_services():
         links=pagination_links(services, '.list_services', request.args))
 
 
-@main.route('/services', methods=['POST'])
-def add_service():
-    message = "IDs are currently generated externally, new resources should " \
-              "be created with PUT"
-    return jsonify(error=message), 501
+@main.route('/services-archive', methods=['GET'])
+def list_archived_services_by_service_id():
+    """
+    Retrieves a list of services from the service_archive table
+    for the supplied service_id
+    :query_param service_id:
+    :return: List[service]
+    """
 
+    try:
+        service_id = int(request.args.get("service-id", "no service id"))
+    except ValueError:
+        abort(400, "Invalid service id supplied")
 
-@main.route('/services/<int:service_id>', methods=['PUT'])
+    try:
+        page = int(request.args.get('page', 1))
+    except ValueError:
+        abort(400, "Invalid page argument")
+
+    services = ServiceArchive.query.filter(Service.service_id == service_id)
+
+    services = services.paginate(page=page, per_page=API_FETCH_PAGE_SIZE,
+                                 error_out=False)
+
+    if request.args and not services.items:
+        abort(404)
+    return jsonify(
+        services=list(map(jsonify_service, services.items)),
+        links=pagination_links(services, '.list_services', request.args))
+
+
+@main.route('/services/<int:service_id>', methods=['POST'])
 def update_service(service_id):
+    """
+        Update a service. Looks service up in DB, and updates the JSON listing.
+        Uses existing JSON Parse routines for validation
+    """
+    service = Service.query.filter(
+        Service.service_id == service_id
+    ).first_or_404()
+
+    service_to_archive = prepare_archived_service(service)
+    update_json = get_json_from_request('update_details')['update_details']
+    validate_updater_json_or_400(update_json)
+    service_update = get_json_from_request('services')['services']
```

Comment:
This is effectively a PATCH request. I totally get that PATCH isn't a well supported HTTP method and therefore using POST against a specific resource URL for partial update is a good alternative. However, I think that the definition and pitfalls described by [RFC5789](http://tools.ietf.org/html/rfc5789) are useful. I think we should document our decision to take this route and commit to using it consistently.